devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,298 @@
+# interlocked
+================
+
+## Overview
+
+`interlocked` exports platform independent atomic operations. It is a platform abstraction and it requires a specific implementation for each platform.
+
+The interface is based on the Windows Interlocked API.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value);
+MOCKABLE_FUNCTION(, int32_t, interlocked_and, volatile int32_t*, destination, int32_t, value);
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_low, int64_t, exchange_high, int64_t, comperand_result);
+MOCKABLE_FUNCTION(, int16_t, interlocked_compare_exchange_16, volatile int16_t*, destination, int16_t, exchange, int16_t, comperand);
+MOCKABLE_FUNCTION(, int64_t, interlocked_compare_exchange_64, volatile int64_t*, destination, int64_t, exchange, int64_t, comperand);
+MOCKABLE_FUNCTION(, void*, interlocked_compare_exchange_pointer, volatile void*, destination, void*, exchange, void*, comperand);
```

Comment:
>volatile void* [](start = 65, length = 14)

I believe here's a missing * somewhere. I think "destination" needs to be a pointer to a volatile pointer, so it can be derefenced and over-written.